src/pb_memory.h

```diff
@@ -57,9 +62,11 @@ class PbMemory {
   static std::unique_ptr<PbMemory> Create(
       std::unique_ptr<SharedMemoryManager>& shm_pool,
       TRITONSERVER_MemoryType memory_type, int64_t memory_type_id,
-      uint64_t byte_size, char* data, bool copy_gpu = true);
+      uint64_t byte_size, char* data, bool copy_gpu = true,
+      bool write_back_data = false, bool copy_data = true);
```

Comment:
why do we need an additional argument for `copy_data`. It looks like it is pretty much similar to `copy_gpu` argument.

To me it looks like `write_back_data` doesn't belong to this class and should be handled outside `PbMemory`.

Reply:
> why do we need an additional argument for copy_data. It looks like it is pretty much similar to copy_gpu argument.

For most of the cases, when creating a new `PbMemory`, we need to copy the data from the original buffer to the newly created buffer using `BackendMemory`. The `copy_data` argument is to handle the case where we just want to prepare the buffer for GPU output and ask the stub process to fill in the data later. Please see here: https://github.com/triton-inference-server/python_backend/pull/293/files#diff-32938d76ce399e799a75fbec5050c99c8546fcc08fb75ab526ea331342731932R310. It is possible that `copy_data` and `copy_gpu` is not aligned, hence using a separate argument here.

> To me it looks like write_back_data doesn't belong to this class and should be handled outside PbMemory. We can discuss more offline.

Makes sense to me. I went ahead and removed `write_back_data` from the `PbMemory` class and handle the data copy outside: https://github.com/triton-inference-server/python_backend/pull/293/files#diff-ca67ec8693bbdfbd82a858e307f87fa39fc9187d39228f1afc5375837dc3f60dR1612.